app/controllers/flowchart_node_controller.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+class FlowchartNodeController < ApplicationController
+  def create
+    prev_id = params[:prev_id]
+    is_child = params[:is_child]
+    begin
+      prev_node = FlowchartNode.find(prev_id)
+    rescue ActiveRecord::RecordNotFound
+      render status: 404, json: { error: "No node found with id #{prev_id}." }
+    else
+      ActiveRecord::Base.transaction do
+        @new_node = FlowchartNode.create(
```

Comment:
Why is this an instance variable on the controller?

Reply:
and in the event that the data is invalid and this create fails, we should be returning the error somewhere?

Reply:
I pushed some changes that address this. I added validation for the node data.